### DISCUSSION TITLE
PC端不能响应触摸事件

### DISCUSSION THREAD

**Discussion by cxh1205** [lsky-org/lsky-pro#386] [问答]
手机端可以正常响应触摸事件，例如点击选中，长按菜单。
但是pc端对触摸事件无法响应，这对于**默认访问pc页面**的平板来说极为不便，需要切换为手机端才可访问。
请问能不能让PC端也响应触摸事件？

**Comment by akatsukiro** [comment]
这应该是你浏览器的问题。我使用的Firefox手机端开启PC模式，正常。
我在我平板上使用默认浏览器（PC模式）访问，正常。

**Reply by cxh1205** [reply]
我平板上用的是华为原装浏览器，开启电脑模式后不能响应触摸事件，但是关闭后可以正常响应，开与关显示的界面是完全相同的。
在电脑上用的是chromium内核的edge浏览器，当设置ua为chrome-pc的时候，可以正常响应鼠标事件但不能响应触摸事件；当ua设置为chrome-mobile phone的时候，可以正常响应鼠标和触摸事件。
更换ua后响应的事件不同，应该不是浏览器的问题，可能是bug。

**Reply by cxh1205** [reply]
可能说的不太清楚，问题是发生在**我的图片**界面，在里面选择图片时，电脑上当ua为pc端，鼠标可以正常勾选图片，但是不能通过触摸选中图片
电脑上ua为安卓或者iphone端，鼠标和触摸都可以正常选中图片

**Reply by 0xxb** [reply]
我的图片页面确实是由客户端ua来识别pc还是手机端，pc和手机端操作不同，不一样。

**Reply by cxh1205** [reply]
原来如此  那手机端目前和电脑端会有什么区别吗 
看上去ua为手机端能力更强  界面和电脑端ua是相同的  但是可以响应触摸和鼠标点击

**Reply by akatsukiro** [reply]
我回去试试。我是三星平板，可能有点不一样的地方。

**Reply by 0xxb** [reply]
> 原来如此 那手机端目前和电脑端会有什么区别吗 看上去ua为手机端能力更强 界面和电脑端ua是相同的 但是可以响应触摸和鼠标点击

区别在于手机端禁用了鼠标拖拽事件来保证点击事件可用，实际上你在pc端点击选择某一项的时候，不是点击事件，而是通过 onmouseup、onmousedown 事件模拟点击，这么做是因为如果同时绑定点击、拖拽事件，事件冒泡顺序是 onmouseup/down -> click ，但是手机端这么做是有问题的，什么问题我不记得了。总之手机端是点击事件，pc端是通过拖拽事件模拟点击事件+拖拽事件。

**Reply by cxh1205** [reply]
那这个能算bug吗，后续有没有修复的可能？

**Reply by akatsukiro** [reply]
啊，确实是有这个问题。我上次好像看的是图片管理，那里触摸好像是没有问题的（因为那里没有右键判定所以能用？）

**Reply by 0xxb** [reply]
等我有空再看看能不能修复

**Reply by cxh1205** [reply]
> 等我有空再看看能不能修复

好的，辛苦你了

**Reply by cxh1205** [reply]
> 啊，确实是有这个问题。我上次好像看的是图片管理，那里触摸好像是没有问题的（因为那里没有右键判定所以能用？）

大概是吧
